### PR TITLE
add searching and ordering

### PR DESCRIPTION
**Pull request by QtiQla**
Hello!
I added two features searching and ordering like [DRF searching filter](https://www.django-rest-framework.org/api-guide/filtering/#searchfilter) and [DRF ordering filter](https://www.django-rest-framework.org/api-guide/filtering/#orderingfilter)
I needed to update your code in pagination.py (sorry about that😊), because property "_ninja_contribute_args" rewrite other wrapped functions.
It look like this

```python
@http_get(path='', response=PaginatedResponseSchema[SomeSchema])
@paginate
@ordering(ordering_fields=['name'])
@searching(search_fields=['name'])
def list(self, filters: SomeFilter = Query(...)):
    return self.get_queryset(filters)
```

glad to help in the development of your project.

Thank you

### CODE REVIEW

**Comment by eadwinCode** [comment]
@QtiQla 
Good job here. I have some concerns to share.
- Your current implementation only covers Synchronous RouteFunctions, you need to cover asynchronous RouteFunctions as well
- Does the ordering and searching depends on pagination? I will like to know if they can exist outside pagination.
- There is no test for this PR or documentation on how to use it.
To fix code style failing tests, you need to do the following:
1) Open the terminal at the root dir of your project
```sh
make install-full
```
That will install all lib. dev dependencies and pre-commit hooks

2)
```sh
make fmt
```
3)
```sh
make lint
```

**Comment by QtiQla** [comment]
- in work
- ordering and searching can exist without pagination, but it is necessary to keep the order of decorators, first we need to filter queryset and then sort queryset. If we use pagination, pagination should be higher than ordering and searching because ordering and searching returns queryset type and pagination returns OrderedDict
- in work, thank you for help!

**Comment by QtiQla** [comment]
Hi there!
@eadwinCode I added async support, tests, and docs for ordering and searching. Can you check it? 
Thank you!

**Comment by eadwinCode** [comment]
@QtiQla Thanks for this good work here. I will review this code this weekend and leave some comments if any.

File: tests/conftest.py [hunk 1]
```diff
@@ -7,6 +7,7 @@ def pytest_configure(config):
     from django.conf import settings
 
     os.environ.setdefault("NINJA_SKIP_REGISTRY", "True")
+    os.environ.setdefault("DJANGO_ALLOW_ASYNC_UNSAFE", "True")
```
Comment (eadwinCode) [inline tests/conftest.py:10]: @QtiQla Please what does this do?

File: docs/tutorial/ordering.md [hunk 2]
```diff
@@ -0,0 +1,83 @@
+# **Ordering**
+
+**Django Ninja Extra** provides an intuitive ordering model using `ordering` decoration from the Django-Ninja-Extra ordering module. It expects a Queryset from as a route function result.
+
+> This feature was inspired by the [DRF OrderingFilter][drf-ordering]
```
Comment (eadwinCode) [inline docs/tutorial/ordering.md:5]: I think you need to add the url for this here `[DRF OrderingFilter](URL)`

File: mkdocs.yml [hunk 3]
```diff
@@ -46,6 +46,8 @@ nav:
   - Form Request: tutorial/form.md
   - Schema: tutorial/schema.md
   - Pagination: tutorial/pagination.md
+  - Ordering: tutorial/ordering.md
```
Comment (eadwinCode) [inline mkdocs.yml:49]: I think this should be described as
```
- Filtering:
   - Ordering: tutorial/ordering.md
   - Searching: tutorial/searching.md
```

**Review by eadwinCode — CHANGES_REQUESTED** [review]
This PR is good to go. Just address the comments and we can have this merged to master. Thanks for your good work here once again.

File: tests/conftest.py (hunk 1 above)
Comment (QtiQla) [reply tests/conftest.py:10]: Oh, my bad. I was not able to create test data in model to test async methods without this variable

File: docs/tutorial/ordering.md (hunk 2 above)
Comment (QtiQla) [reply docs/tutorial/ordering.md:5]: np

File: mkdocs.yml (hunk 3 above)
Comment (QtiQla) [reply mkdocs.yml:49]: i'll change it

File: tests/conftest.py (hunk 1 above)
Comment (eadwinCode) [reply tests/conftest.py:10]: Is it still useful? if not you can remove it.

Comment (QtiQla) [reply tests/conftest.py:10]: Yes, it is. Without this var async tests  doesn't work, they get exeption like "You cannot call this from an async context - use a thread or sync_to_async", i couldn't fix it without using the variable above

**Comment by QtiQla** [comment]
@eadwinCode  please don't release this version, because i made mistake for async version. I faced [common GOTCHA](https://django-ninja.rest-framework.com/guides/async-support/) - "Django queryset's are lazily evaluated (database query happens only when you start iterating), so this will not work", I'm preparing version with list support for async version. Sorry about that

**Pull request merged**